inc/Server.hpp

```diff
@@ -105,6 +107,8 @@ class Server {
 	time_t m_clientTimeout; /**< Timeout for a Connection in seconds */
 	std::map<int, Socket> m_virtualServers; /**< Listening sockets of virtual servers */
 	std::map<int, Connection> m_connections; /**< Current active Connections */
+	std::vector<char> m_clientHeaderBuffer; /**< Buffer for reading request header */
```

Comment:
Why have it as a member instead of just a local variable as before the char[]?

Reply:
The char[] can be initiated on the stack. 

In C++98 you can't initiate a std::vector on the stack, it always allocates on the heap. To avoid allocations everytime we enter this function I created member variable